File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.swift
import Foundation

func solution(_ numbers:[Int], _ target:Int) -> Int {
    let ret: Int = go(0, 0, numbers, target)
    return ret
}

func go(_ index: Int, _ sum: Int, _ numbers: [Int], _ target: Int) -> Int {
    if index == (numbers.count) {
        if sum == target {
            return 1
        }
        return 0
    }
    
    let a = sum + numbers[index]
    let b = sum - numbers[index]
    
    return go(index + 1, a, numbers, target) + go(index + 1, b, numbers, target)
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <bits/stdc++.h>

using namespace std;

// 2^20
int cnt;

void dfs(int idx, int sum, vector<int>& numbers, int target) {
    if(idx == (int)numbers.size()) {
        if(sum == target) cnt++;
        return;
    }
    
    dfs(idx + 1, sum + numbers[idx], numbers, target);
    dfs(idx + 1, sum - numbers[idx], numbers, target);
}

int solution(vector<int> numbers, int target) {
    cnt = 0;
    dfs(0, 0, numbers, target);
    return cnt;
}